lib/routes/caixin/utils-fulltext.ts

```diff
@@ -0,0 +1,46 @@
+import crypto from 'crypto';
+import { hextob64, KJUR } from 'jsrsasign';
+import ofetch from '@/utils/ofetch';
+import { config } from '@/config';
+
+const rsaPrivateKey =
```

Comment:
could we generate the rsa private key using method like <https://nodejs.org/api/crypto.html#cryptogeneratekeypairtype-options-callback> instead of hardcode it?

Reply:
This key is not any random key. It is used to produce the signature to be verified by caixin's server. So it must be the pair to the public key that is held by the server.

I've considered parsing it from https://file.caixin.com/pkg/cx-pay-layer/js/wap.js?v=5.15.421933 automatically, but locating and extracting necessary code that computes it is rather hard.

Reply:
where can we verify the public key on the caixin server? Can we provide a instruction fir user to set their own private key in the Environment Variable?

Reply:
> where can we verify the public key on the caixin server?

> Can we provide a instruction fir user to set their own private key in the Environment Variable?

 My previous explanation might have not been very clear. The private key is not a per-user string, from my understanding. It is an intermediate variable calculated during the execution of the script from the link above, which I believe is universally the same.

Then, while I can share my steps on getting the key from the script, it involves using the browser devtools, which is a bit complex for general users in my opinion.

Reply:
If the RSA private key is globally static and contains no sensitive information, hardcoding it in the code is acceptable but should be clearly commented/documented.

May you add more comments of this line?

Reply:
Will do. Give me a moment.